pkg/server/home_workspaces.go

```diff
@@ -0,0 +1,664 @@
+
+		responsewriters.WriteObjectNegotiated(homeWorkspaceCodecs, negotiation.DefaultEndpointRestrictions, tenancyv1beta1.SchemeGroupVersion, rw, req, http.StatusOK, homeWorkspace)
+		return
+	}
+
+	needsAutomaticCreation, workspaceType := h.needsAutomaticCreation(lcluster.Name)
+	if !needsAutomaticCreation {
+		h.apiHandler.ServeHTTP(rw, req)
+		return
+	}
+
+	isHome := workspaceType == "Home"
+	if foundLocally, retryAfterSeconds, err := h.searchForReadyWorkspaceInLocalInformers(lcluster.Name, isHome, user.GetName()); foundLocally {
+		if retryAfterSeconds > 0 {
+			// Return a 429 status asking the client to try again after the creationDelay
+			rw.Header().Set("Retry-After", fmt.Sprintf("%d", retryAfterSeconds))
+			http.Error(rw, "Creating the home workspace", http.StatusTooManyRequests)
+			return
+		}
+		h.apiHandler.ServeHTTP(rw, req)
+		return
+	} else if err != nil {
+		responsewriters.InternalError(rw, req, err)
+		return
+	}
+
+	// Home or bucket workspace not found in the local informer
+	// Let's try to create it
+
+	// But first check we have the right to do so.
+	attributes := homeWorkspaceAuthorizerAttributes(user, "create")
+
+	if isHome && lcluster.Name != h.getHomeLogicalClusterName(user.GetName()) {
+		// If we're checking a home workspace or home bucket workspace, but not of the consistent user, let's refuse.
+		klog.Errorf("failed to authorize user %q to create a home workspace %q: home workspace can only be created by the user of the home workspace", user.GetName(), lcluster.Name)
+		responsewriters.Forbidden(ctx, attributes, rw, req, authorization.WorkspaceAcccessNotPermittedReason, homeWorkspaceCodecs)
+		return
+	}
+
+	// Test if the user has the right to create his Home workspace when it doesn't exists
+	// => test the create verb on the clusterworkspaces/workspace subresource named ~ in the root workspace.
+
+	if decision, reason, err := h.authz.Authorize(
+		request.WithCluster(ctx, request.Cluster{Name: tenancyv1alpha1.RootCluster}),
+		attributes); err != nil {
+		responsewriters.InternalError(rw, req, err)
+		return
+	} else if decision != authorizer.DecisionAllow {
+		responsewriters.Forbidden(ctx, attributes, rw, req, reason, homeWorkspaceCodecs)
+		return
+	}
+
+	if retryAfterSeconds, err := h.tryToCreate(ctx, user.GetName(), lcluster.Name, workspaceType); err != nil {
+		responsewriters.ErrorNegotiated(err, errorCodecs, schema.GroupVersion{}, rw, req)
+		return
+	} else {
+		// Return a 429 status asking the client to try again after the creationDelay
+		rw.Header().Set("Retry-After", fmt.Sprintf("%d", retryAfterSeconds))
+		http.Error(rw, "Creating the home workspace", http.StatusTooManyRequests)
+		return
+	}
+}
+
+// reHomeWorkspaceNameDisallowedtChars is the regexp that defines what characters
+// are disallowed in a home workspace name.
+// Home workspace name is derived from the user name, with disallowed characters
+// replaced.
+var reHomeWorkspaceNameDisallowedtChars = regexp.MustCompile("[^a-z0-9-]")
```

Comment:
```suggestion
var reHomeWorkspaceNameDisallowedChars = regexp.MustCompile("[^a-z0-9-]")
```

Reply:
Fixed in commit ae9862f8e0de72cb7366e5289b4991dda8a1f72e